test/unit/range/iterator_test_template.hpp

```diff
@@ -65,58 +76,61 @@ TYPED_TEST_P(iterator_fixture, concept_check)
 {
     using iterator_type = decltype(std::ranges::begin(this->test_range));
     // Ensure that reference types are comparable if no equal_eq function was defined.
-    if constexpr (!HasExpectEqualMemberFunction<iterator_fixture<TypeParam>>)
+    if constexpr (!has_expect_equal_member_function<iterator_fixture<TypeParam>>)
     {
         static_assert(std::equality_comparable_with<decltype(*std::ranges::begin(this->test_range)),
                                                     decltype(*std::ranges::begin(this->expected_range))>,
                       "The reference types of begin_iterator and expected_range must be equality comparable. "
                       "If they are not, you may specify a custom void expect_eq(i1, r2) function in the fixture.");
     }
 
-    if constexpr (std::same_as<typename TestFixture::iterator_tag, std::input_iterator_tag>)
-    {
-        static_assert(std::input_iterator<decltype(std::ranges::begin(this->expected_range))>,
-                      "expected_range must have a begin member function and "
-                      "the returned iterator must model std::input_iterator.");
-        EXPECT_TRUE(std::input_iterator<iterator_type>);
-    }
-    else if constexpr (std::same_as<typename TestFixture::iterator_tag, std::forward_iterator_tag>)
+    // input iterator must always be satisfied
+    static_assert(std::ranges::input_iterator<decltype(std::ranges::begin(this->expected_range))>,
+                  "expected_range must have a begin member function and "
+                  "the returned iterator must model std::input_iterator.");
+    EXPECT_TRUE(std::ranges::input_iterator<iterator_type>);
+
+    if constexpr (at_least_forward_iterator_tag<typename TestFixture::iterator_tag>)
     {
-        static_assert(std::forward_iterator<decltype(std::ranges::begin(this->expected_range))>,
+        static_assert(std::ranges::forward_iterator<decltype(std::ranges::begin(this->expected_range))>,
                       "expected_range must have a begin member function and "
-                      "the returned iterator must model std::forward_iterator.");
-        EXPECT_TRUE(std::forward_iterator<iterator_type>);
+                      "the returned iterator must model std::ranges::forward_iterator.");
+        EXPECT_TRUE(std::ranges::forward_iterator<iterator_type>);
```

Comment:
```suggestion
        EXPECT_TRUE(std::forward_iterator<iterator_type>);
```

This is OUR fuck up that `std::ranges` also contains `std::forward_iterator`,  `std::default_sentinel` and so on.

The rule is: Iterator related things are in `std::`, ranges related things are in `std::ranges`

To remove those things from `std::ranges` is part of https://github.com/seqan/product_backlog/issues/49